workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactory.kt

```diff
@@ -5,85 +5,162 @@ import android.view.View
+ * Use the [ScreenViewFactoryFinder] in [environment] to return the [ScreenViewFactory]
+ * bound to the type of the receiving [Screen].
  *
- * @param viewStarter An optional wrapper for the function invoked when [View.start]
- * is called, allowing for last second initialization of a newly built [View].
- * See [ViewStarter] for details.
- *
- * @throws IllegalArgumentException if no builder can be found for type [ScreenT]
+ * - It is more common to use [WorkflowViewStub.show] than to call this method directly
+ * - Call [ScreenViewFactory.start] to create and initialize a new [View]
+ * - If you don't particularly need to mess with the [ScreenViewFactory] before creating
+ *   a view, use [Screen.toView] instead of this method.
+ */
+@WorkflowUiExperimentalApi
+public fun <ScreenT : Screen> ScreenT.toViewFactory(
+  environment: ViewEnvironment
+): ScreenViewFactory<ScreenT> {
+  return environment[ScreenViewFactoryFinder].getViewFactoryForRendering(environment, this)
+}
+
+/**
+ * Creates a [ScreenViewHolder] wrapping a [View] able to display [initialRendering],
+ * and initializes the view.
  *
- * @throws IllegalStateException if the matching [ScreenViewFactory] fails to call
- * [View.bindShowRendering] when constructing the view
+ * By default "initialize" means making the first call to [ScreenViewHolder.show].
+ * To add more initialization behavior (typically a call to [WorkflowLifecycleOwner.installOn]),
+ * provide a [viewStarter].
  */
+@Suppress("DEPRECATION")
 @WorkflowUiExperimentalApi
-public fun <ScreenT : Screen> ScreenT.buildView(
-  viewEnvironment: ViewEnvironment,
+public fun <ScreenT : Screen> ScreenViewFactory<ScreenT>.start(
+  initialRendering: ScreenT,
+  initialViewEnvironment: ViewEnvironment,
   contextForNewView: Context,
   container: ViewGroup? = null,
-  viewStarter: ViewStarter? = null,
-): View {
-  val viewFactory = viewEnvironment[ScreenViewFactoryFinder].getViewFactoryForRendering(
-    viewEnvironment, this
-  )
+  viewStarter: ViewStarter? = null
+): ScreenViewHolder<ScreenT> {
+  return ScreenViewHolder(
+    this,
+    initialViewEnvironment,
+    initialRendering,
+    buildView(initialViewEnvironment, contextForNewView, container),
+  ).also { holder ->
+    val resolvedStarter = viewStarter ?: ViewStarter { _, doStart -> doStart() }
 
-  return viewFactory.buildView(this, viewEnvironment, contextForNewView, container).also { view ->
-    checkNotNull(view.workflowViewStateOrNull) {
-      "View.bindShowRendering should have been called for $view, typically by the " +
-        "ScreenViewFactory that created it."
-    }
-    viewStarter?.let { givenStarter ->
-      val doStart = view.starter
-      view.starter = { newView ->
-        givenStarter.startView(newView) { doStart.invoke(newView) }
+    val legacyStarter: ((View) -> Unit)? = holder.view.starterOrNull
+
+    if (legacyStarter != null) {
+      var shown = false
```

Comment:
nit:  This variable is duplicated in the `else` block and could be pulled up out of the `if` statement.

Reply:
That was actually intentional, b/c I wanted to customize the exception message.

Reply:
Like, the duplication felt cleaner than letting each `shown` bleed into a wider scope.

Reply:
Got it.